taca/cleanup/cleanup.py

```diff
@@ -0,0 +1,644 @@
+                proj_info['fastq_size'] = 0
+                project_clean_list[proj_info['name']] = proj_info
+    else:
+        for flowcell_dir in flowcell_dir_root:
+            for fc in [d for d in os.listdir(flowcell_dir) if re.match(filesystem.RUN_RE,d)]:
+                fc_abs_path = os.path.join(flowcell_dir, fc)
+                with filesystem.chdir(fc_abs_path):
+                    projects_in_fc = [d for d in os.listdir(flowcell_project_source) \
+                                      if re.match(r'^[A-Z]+[_\.]+[A-Za-z]+_\d\d_\d\d$',d) and \
+                                      not os.path.exists(os.path.join(flowcell_project_source, d, "cleaned"))]
+                    for _proj in projects_in_fc:
+                        proj = re.sub(r'_+', '.', _proj, 1)
+                        if proj in project_processed_list:
+                            if proj in project_clean_list and project_clean_list[proj]['closed_days'] >= days_fastq:
+                                fc_fq_files, fq_size = collect_fastq_data_irma(fc_abs_path, os.path.join(flowcell_project_source, _proj), return_size=True)
+                                project_clean_list[proj]['fastq_to_remove']['flowcells'][fc] = fc_fq_files['flowcells'][fc]
+                                project_clean_list[proj]['fastq_size'] += fq_size
+                            continue
+                        project_processed_list.append(proj)
+                        #by default assume all projects are not old enough for delete
+                        fastq_data, analysis_data = ("young", "young")
+                        fastq_size, analysis_size = (0, 0)
+                        proj_info = get_closed_proj_info(pcon.get_entry(proj))
+                        if proj_info:
+                            # if project not old enough for fastq files and only fastq files selected move on to next project
+                            if proj_info['closed_days'] >= days_fastq:
+                                fastq_data, fastq_size = collect_fastq_data_irma(fc_abs_path, os.path.join(flowcell_project_source, _proj),
+                                                                                 data_dir, proj_info['pid'], return_size=True)
+                            if not only_fastq:
+                                # if project is old enough for fastq files and not 'only_fastq' try collect analysis files 
+                                if proj_info['closed_days'] >= days_analysis:
+                                    analysis_data, analysis_size = collect_analysis_data_irma(proj_info['pid'], analysis_dir,
+                                                                                              analysis_data_to_remove, return_size=True)
+                                # if both fastq and analysis files are not old enough move on
+                                if (analysis_data == fastq_data) or ((not analysis_data or analysis_data == "cleaned") and fastq_data == "young"):
```

Comment:
I do not follow here... `analysis_data` is initialised to `young` and is assigned it becomes  dict . It is None only if the path does not exits but in that case we are in serious troubles... so the `not analysis_data` should always be evaluated to `False`